bundle-workflow/src/test_workflow/test_recorder.py

```diff
@@ -3,34 +3,133 @@
 # The OpenSearch Contributors require contributions made to
 # this file be licensed under the Apache-2.0 license or a
 # compatible open source license.
-
 import logging
+import os
+import shutil
+import tempfile
+
+import yaml
 
 
 class TestRecorder:
-    def __init__(self, location):
-        self.location = location
-        logging.info(f"TestRecorder storing results in {location}")
+    ACCEPTED_TEST_TYPES = ["integ-test", "bwc-test", "perf-test"]
+
+    def __init__(self, test_run_id, test_type, location=None):
+        self.test_type = test_type
+        self.test_run_id = test_run_id
+
+        if self.test_type not in self.ACCEPTED_TEST_TYPES:
+            raise ValueError(f"test_type is invalid. Acceptable test_type are {self.ACCEPTED_TEST_TYPES}")
+
+        if location is None:
+            location = tempfile.TemporaryDirectory()
+            self.location = location.name
+        else:
+            self.location = location
+        logging.info(f'TestRecorder storing results in {os.path.realpath(self.location)}')
+
+    def record_local_cluster_logs(self, component_name, component_test_config, stdout, stderr, log_files):
+        """
+        Record the test cluster logs.
+        :param component_name: component that is under test right now.
+        :param component_test_config: component_config under consideration for test eg: with/without-security.
+        :param stdout: A string containing the stdout stream from the test process.
+        :param stderr: A string containing the stderr stream from the test process.
+        :param log_files: A generator that yields tuples containing test cluster log files, in the form (absolute_path, relative_path).
+        """
 
-    def record_cluster_logs(self, log_files):
+        base = self.__create_base_folder_structure(component_name, component_test_config)
+        dest_directory = os.path.join(base, "local_cluster_logs")
+        os.makedirs(dest_directory, exist_ok=False)
+        logging.info(
+            f"Recording local cluster logs for {component_name} with {component_test_config} config in {os.path.realpath(dest_directory)}")
+        self.__generate_std_files(stdout, stderr, os.path.realpath(dest_directory))
+        local_cluster_log_files = list(log_files)
+        for log_file in local_cluster_log_files:
+            dest_file = os.path.join(dest_directory, os.path.basename(log_file[0]))
+            shutil.copyfile(log_file[0], dest_file)
+
+    def record_remote_cluster_logs(self, component_name, component_test_config, exit_code, stdout, stderr,
```

Comment:
The flow doesn't clean to me. Let me know if I didn't understand this correctly.

- For test suite which uses local cluster, it uses both `record_test_outcome` and `record_local_cluster_logs`.
   `record_test_outcome` generates the test outcome manifest.
- For test suite which uses remote cluster, it just uses `record_remote_cluster_logs `. `record_remote_cluster_logs` generates the test outcome manifest.

If that is the case why don't we merge both remote and local test cluster into one function `record_test_cluster` which takes in local/remote as a parameter. The only logic different between them is creating a different directory.

Let me know what you think.

Reply:
I agree. I'll work on this. One more logic that differs is remote cluster logs won't have log_files but just the location of the logs. Will see how to differentiate that.

Reply:
`record_local_cluster_logs` also copies the local test cluster log files into the directory where remote cluster just gives the location of the logs. That will be an additional change but with right param it should be distinguishable